src/Functions/materialize.h

```diff
@@ -9,13 +9,14 @@ namespace DB
 
 /** materialize(x) - materialize the constant
   */
+template <bool remove_sparse>
```

Comment:
Is it really worth adding the flag as a template parameter? An extra `if` at runtime per each `executeImpl` call (which is virtual also) should not affect performance.

Reply:
I did it because `FunctionFactory::registerFunction` receives the function as a template parameter and I don't see how to pass this flag there in the other way.

Reply:
`registerFunction` can accept lambda as well https://github.com/ClickHouse/ClickHouse/blob/f0f9224d6aa9528222dac51afb9d109141656ee7/src/Functions/FunctionFactory.h#L55-L65